core/src/Cuda/Kokkos_Cuda_Parallel_Range.hpp

```diff
@@ -796,7 +797,9 @@ class ParallelScanWithTotal<FunctorType, Kokkos::RangePolicy<Traits...>,
   size_type* m_scratch_space;
   size_type* m_scratch_flags;
   size_type m_final;
-  ReturnType& m_returnvalue;
```

Comment:
Seeing that `ReturnType` is no longer in use anywhere within the class including other backends, is it still needed in the template parameter and partial specialization list?

Reply:
It's useful when we want the return type to determine the value type, e.g., to allow the functor's call operator to be templated on the index type, see #5169.

Reply:
Ok. I can see the difficulty of always expecting to have `FunctorAnalysis` available. In that case, for the consistency, I'm also okay leaving this as is if #5169 gets merged in as well.